src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -19,32 +21,68 @@
       DetermineProjectJsonPath;
       ResolveNuGetPackages;
     </ResolveAssemblyReferencesDependsOn>
+    <ResolveNuGetPackagesDependsOn>
+      $(ResolveNuGetPackagesDependsOn);
+      DetermineProjectJsonPath;
+    </ResolveNuGetPackagesDependsOn>
   </PropertyGroup>
 
   <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherAssemblyVersionsFromPackagesDrops;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
-
-  <!-- Override targets for building against packages -->
-  <Target Name="BuildAndTest" DependsOnTargets="GenerateTestProjectJson;Build;Test" />
-  <Target Name="RebuildAndTest" DependsOnTargets="GenerateTestProjectJson;Rebuild;Test" />
-
-  <Target Name="DetermineProjectJsonPath">
+  
+  <Target Name="DetermineProjectJsonPath"
+          DependsOnTargets="EvaluateBuildsProjects">
     <PropertyGroup>
       <OriginalProjectJson>$(ProjectJson)</OriginalProjectJson>
+      <GeneratedTargetGroup Condition="'$(GeneratedTargetGroup)' == ''">$(TargetGroup)</GeneratedTargetGroup>
+      <GeneratedOSGroup Condition="'$(GeneratedOSGroup)' == ''">$(OSGroup)</GeneratedOSGroup>
       <!-- We can't put the generated project.json in the output path or NuGet will interpret Microsoft.Win32.Primitives as a "project"
            type instead of a "package" type when generating the project.lock.json file, so we have to place our project.json in a different path location. -->
-      <ProjectJson>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\json\project.json</ProjectJson>
-      <GeneratedProjectLockJson>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\json\project.lock.json</GeneratedProjectLockJson>
+      <ProjectsGeneratedProjectJsonDir>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\</ProjectsGeneratedProjectJsonDir>
+      <ProjectsGeneratedProjectJsonDir Condition="'$(GeneratedOSGroup)' != ''">$(ProjectsGeneratedProjectJsonDir)$(GeneratedOSGroup)\</ProjectsGeneratedProjectJsonDir>
+      <ProjectsGeneratedProjectJsonDir Condition="'$(GeneratedTargetGroup)' != ''">$(ProjectsGeneratedProjectJsonDir)$(GeneratedTargetGroup)\</ProjectsGeneratedProjectJsonDir>
+      <ProjectJson>$(ProjectsGeneratedProjectJsonDir)project.json</ProjectJson>
+      <GeneratedProjectLockJson>$(ProjectsGeneratedProjectJsonDir)project.lock.json</GeneratedProjectLockJson>
       <!-- Not all projects generate a new project.json, so only update the project.lock.json if it has been restored. -->
       <ProjectLockJson Condition="Exists('$(GeneratedProjectLockJson)')">$(GeneratedProjectLockJson)</ProjectLockJson>
     </PropertyGroup>
     <Message Condition="Exists('$(ProjectLockJson)')" Text="Using '$(ProjectLockJson)'" Importance="Low"/>
   </Target>
 
+  <!-- The msbuild project file may be an ".builds" file, if it is, then we need to evaluate the "Project" items in the builds file
+       to generate the project.json's for each of those items. -->
+  <Target Name="EvaluateBuildsProjects"
```

Comment:
The entire point of dir.traversals.targets is to iterate on the ".builds" files. So I guess I'm still missing the point of why we are doing it here and there. So what is the difference between this and https://github.com/dotnet/corefx/pull/7604/files#diff-cd0fc5e0bad8102e1a45aa7575bdd102R175.